src/Illuminate/Foundation/Testing/Concerns/InteractsWithRedis.php

```diff
@@ -27,6 +28,7 @@
      */
     public function setUpRedis()
     {
+        $app = $this->app ?? new Application;
```

Comment:
This 💩 is necessary, because `InteractsWithRedis` is used by integration tests (which have an application instance) and by unit tests (which do not have an application instance).

Reply:
should it be `$app = $this->app ?? resolve(Application::class);` so we can mock out or inject the Application during tests if needed?

Reply:
I don't see `resolve()` being used anywhere else in the tests, so I rather not do that. Can you explain why you want to start using a helper class here?

Reply:
...well - same as `new RedisManager` that I just noticed further down. By hard coding them, you can never use Mockey on either of those.

And if you were ever binding them with a custom version in your application, then this would not resolve the custom version from the container.

However, as I type this - I guess that if you dont have `$this->app` - then your running a unit test, in which case it probably doesnt matter...

Reply:
Can you resolve the container from the container? (In reference to `resolve(Application::class);`)

Reply:
lol - probably not. silly me.

But the redismanager you could.